### PR TITLE
pkp/pkp-lib#685 store salt in a file and better information for the r…

### PR DESCRIPTION
**Pull request by bozana**
…eader

The salt can now be saved in a file. The information for the users/readers is improved.

### CODE REVIEW

File: plugins/generic/usageStats/UsageStatsPlugin.inc.php [hunk 1]
```diff
@@ -306,9 +306,42 @@ function getUsageEventCurrentDayLogName() {
...
+				} else {
+					$newSalt = mt_rand();
+				}
+				$file = fopen($saltFilename,'wb');
+				if (flock($file, LOCK_EX)) {
+					fwrite($file, $newSalt);
+					flock($file, LOCK_UN);
+				} else {
+					// Couldn't lock the file.
+					assert(false);
```
Comment (beghelli) [inline plugins/generic/usageStats/UsageStatsPlugin.inc.php]: maybe you can notify this error? Just use the ScheduledTask::addExecutionEntry method...

File: plugins/generic/usageStats/UsageStatsPlugin.inc.php [hunk 2]
```diff
@@ -306,9 +306,42 @@ function getUsageEventCurrentDayLogName() {
...
+				}
+				fclose($file);
+				$salt = $newSalt;
+			}
+		}
+
+		// Manage the IP address (evtually hash it)
+		if ($this->getSetting(CONTEXT_ID_NONE, 'dataPrivacyOption')) {
+			if (!isset($salt)) return false;
```
Comment (beghelli) [inline plugins/generic/usageStats/UsageStatsPlugin.inc.php]: add another entry into the execution log maybe?

**Comment by beghelli** [comment]
reviewed Bozana, just minor comments about logging the errors. :)

**Comment by bozana** [comment]
@beghelli: like this?
THANKS!!!

File: plugins/generic/usageStats/UsageStatsPlugin.inc.php [hunk 3]
```diff
@@ -306,9 +306,44 @@ function getUsageEventCurrentDayLogName() {
...
+					$newSalt = bin2hex(file_get_contents('/dev/urandom', false, null, 0, 16));
+				} else {
+					$newSalt = mt_rand();
+				}
+				$file = fopen($saltFilename,'wb');
+				if (flock($file, LOCK_EX)) {
+					fwrite($file, $newSalt);
+					flock($file, LOCK_UN);
+				} else {
+					ScheduledTask::addExecutionLogEntry(__('plugins.generic.usageStats.saltWriteError', array()), SCHEDULED_TASK_MESSAGE_TYPE_ERROR);
```
Comment (beghelli) [inline plugins/generic/usageStats/UsageStatsPlugin.inc.php]: I don't think you need to pass the empty array when calling the translate method...

File: plugins/generic/usageStats/UsageStatsPlugin.inc.php [hunk 4]
```diff
@@ -306,9 +306,44 @@ function getUsageEventCurrentDayLogName() {
...
+				}
+				fclose($file);
+				$salt = $newSalt;
+			}
+		}
+
+		// Manage the IP address (evtually hash it)
+		if ($this->getSetting(CONTEXT_ID_NONE, 'dataPrivacyOption')) {
+			if (!isset($salt)) {
+				ScheduledTask::addExecutionLogEntry(__('plugins.generic.usageStats.hashSaltMissing', array()), SCHEDULED_TASK_MESSAGE_TYPE_ERROR);
```
Comment (beghelli) [inline plugins/generic/usageStats/UsageStatsPlugin.inc.php]: I don't think that the addExecutionLogEntry is static. You can use $this->addExecutionLogEntry() instead

Comment (ctgraham) [inline plugins/generic/usageStats/UsageStatsPlugin.inc.php]: But UsageStatsPlugin doesn't extend ScheduledTask, right?  So `$this` won't have access to the method, I think.

Comment (beghelli) [inline plugins/generic/usageStats/UsageStatsPlugin.inc.php]: yes, Bozana already taught me that lesson. :) Thanks Clinton.

**Comment by bozana** [comment]
I removed the scheduled tasks logging on this place again... :-)

**Comment by bozana** [comment]
Sorry, I've made some HTML corrections... :-)

**Comment by beghelli** [comment]
Thanks Bozana, once the CI is complete, we can merge. :)

**Comment by bozana** [comment]
Hi Bruno, I've fixed the remained issues... THANKS A LOT!!!

**Comment by bozana** [comment]
Hi Bruno, I've rebased the branch with the ojs-dev-2_4 -- the merge conflicts should be solved now... :-)

**Pull request merged**